### DISCUSSION TITLE
[NextJS] Where to configureScope ?

### DISCUSSION THREAD

**Discussion by ahetawal-p** [getsentry/support#89] [Support] [answered]
Hello,
I am working through setting up sentry for our NextJS app. 
I am curious what is best way to configure scope for our sentry setup specifically, `user` info. 
We only get user info once the user logs into the app, and they might not need to login again if already logged in. 
So what would be the best place to configure the scope, so I dont have to do it on every nextjs page ?

**Comment by souredoutlook** [comment]
Hi @ahetawal-p - It is possible to initialize Sentry with [initialScope](https://docs.sentry.io/platforms/javascript/guides/nextjs/configuration/options/#initial-scope) which might suit your needs.

**Reply by ahetawal-p** [reply]
Hmm, I thought `initialScope` is setup in during `client.config.js` or `server.config.js` and we dont have the auth user info at that time based, on the how the firebase authentication is integrated in nextjs.
And also, even if we are able to set this up like above, what if user logs out after the launching the app, is there a way to reset this scope then ?

**Comment by souredoutlook** [comment, marked as answer]
@ahetawal-p given that you're using Firebase you'll likely want to go with [configureScope](https://docs.sentry.io/platforms/javascript/guides/nextjs/enriching-events/scopes/#configuring-the-scope) in whatever hooks you're using to check authentication state.

**Reply by ahetawal-p** [reply]
Got it, yes thats what I ended up doing. Thank you again.
